pacsman/base_client.py

```diff
@@ -42,14 +42,19 @@ def verify(self) -> bool:
         raise NotImplementedError()
 
     @abstractmethod
-    def search_patients(self, search_query: str, additional_tags: List[str] = None,
+    def search_patients(self, search_query: Optional[str] = None,
+                        search_query_type: Optional[str] = None,
+                        additional_tags: Optional[List[str]] = None,
                         wildcard: bool = True) -> List[Dataset]:
         """
         Search for patients. The PatientID and PatientName are searched.
-        Performs a partial match.
-        :param search_query: Search string for either patient name or ID
+        Performs a partial match if wildcard is True.
+        :param search_query: String containing query value to c_find.
+        :param search_query_type: Optional string that restricts patient search to 'PatientID' or 'PatientName'.
```

Comment:
I like your thinking about pushing this improvement up a layer of abstraction to the base class. However, we probably cannot do this right now because no other backends implement handling for this `search_query_type` parameter besides the DCMTK backend. For now, let's move this back to the DCMTK client level, and create a follow-up issue for extending this to all pacsman backends.

Reply:
Ok, sounds good.